dask_expr/_collection.py

```diff
@@ -662,10 +668,19 @@ def shift(self, periods=1, freq=None, axis=0):
             raise TypeError("periods must be an integer")
 
         axis = _validate_axis(axis)
-        if axis == 1:
-            raise NotImplementedError("shift on axis 1 not supported yet")
-
-        return new_collection(Shift(self.expr, periods, freq))
+        if axis == 0:
+            return new_collection(Shift(self.expr, periods, freq))
+
+        clear_divisions = _calc_maybe_new_divisions(self.expr, periods, freq) is None
```

Comment:
We generally want to avoid accessing the divisions as long as possible. Can we just set those to None for now and add a todo?